src/features/emails/components/EmailEditor/tools/inlineVariable/index.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  API,
+  InlineTool,
+  InlineToolConstructorOptions,
+} from '@editorjs/editorjs';
+
+import InlineToolBase from '../../utils/InlineToolBase';
+
+export default function variableToolFactory(title: string) {
+  class VariableTool extends InlineToolBase implements InlineTool {
+    private _api: API;
+    private _availableVars: { [slug: string]: string };
+    private _button: HTMLButtonElement;
+    private _list: HTMLUListElement;
+    private _pendingRange: Range | null = null;
+
+    constructor({ api }: InlineToolConstructorOptions) {
+      super();
+
+      this._api = api;
+
+      this._availableVars = {
+        first_name: 'First name',
+        full_name: 'Full name',
+        last_name: 'Last name',
+      };
+
+      this._button = document.createElement('button');
+      this._button.type = 'button';
+      const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+      svg.setAttributeNS(null, 'viewBox', '0 0 24 24');
+      svg.style.display = 'block';
+      svg.style.padding = '2px';
+      const path = document.createElementNS(
+        'http://www.w3.org/2000/svg',
+        'path'
+      );
+      path.setAttributeNS(
+        null,
+        'd',
+        'M4 7v2c0 .55-.45 1-1 1H2v4h1c.55 0 1 .45 1 1v2c0 1.65 1.35 3 3 3h3v-2H7c-.55 0-1-.45-1-1v-2c0-1.3-.84-2.42-2-2.83v-.34C5.16 11.42 6 10.3 6 9V7c0-.55.45-1 1-1h3V4H7C5.35 4 4 5.35 4 7m17 3c-.55 0-1-.45-1-1V7c0-1.65-1.35-3-3-3h-3v2h3c.55 0 1 .45 1 1v2c0 1.3.84 2.42 2 2.83v.34c-1.16.41-2 1.52-2 2.83v2c0 .55-.45 1-1 1h-3v2h3c1.65 0 3-1.35 3-3v-2c0-.55.45-1 1-1h1v-4z'
+      );
+      path.setAttributeNS(null, 'stroke-width', '0');
+      svg.appendChild(path);
+      this._button.appendChild(svg);
+      this._button.classList.add(this._api.styles.inlineToolButton);
+
+      this._list = document.createElement('ul');
+      this._list.style.display = 'none';
+      this._list.style.listStyleType = 'none';
+      this._list.style.minWidth = '150px';
+      this._list.style.padding = '0px';
+      this._list.style.margin = '0px';
+
+      Object.entries(this._availableVars).forEach(([slug, title]) => {
+        const item = document.createElement('li');
+        item.style.cursor = 'pointer';
+        item.style.padding = '8px';
+        item.style.borderTop = '1px solid rgba(0,0,0,0.1)';
+        item.addEventListener('mouseenter', () => {
+          item.style.backgroundColor = 'rgba(0,0,0,0.05)';
+        });
```

Comment:
these are not theme colors, just hard coded. does it maybe make sense here to take in colours from the theme instead of doing "transparent black" here? I could also be fine with it being hard coded here.

Reply:
but this is for the drop down menu of the variable options. it's code i made myself, so i am to blame for it lol. i agree that it suits the background of a variable in the editor to be translucent black! but let's leave it and correct later if we feel it becomes a problem with the local email themes of the orgs.